devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -56,26 +65,102 @@ metadata {
+        fingerprint mfr: "0312", prod: "A000", model: "A005", deviceJoinName: "Evalogik Switch" //ZW30
+        fingerprint mfr: "0312", prod: "BB00", model: "BB01", deviceJoinName: "Evalogik Switch" //ZW30S Evalogik Smart on/off Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB03", deviceJoinName: "Evalogik Switch" //ZW30TS Evalogik Smart on/off Toggle Switch
     }
 
     preferences {
-        configParams.each {
-            if (it.name) {
-                if (it.range) {
-                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-                } else {
-                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-                }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Paddle Control' and 'createButton' are only valid for the devices with product number of MS10ZS, MS12ZS, ZW30, ZW30S, ZW30TS(one of them)",
+                required: false
+        getConfigParamInput(paddleControlParam)
+        input(type: "enum", name: "createButton", required: false, title: "Create Button for Paddles?", options: ["No", "Yes"], defaultValue:"Yes")
+    }
+}
+
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private initialize() {
+    if (device.latestValue("checkInterval") != checkInterval) {
+        sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+    }
+    if (isButtonAvailable()) {
+        state.createButtonEnabled = (safeToInt(settings?.createButton) != 0)
+        if (state.createButtonEnabled && !childDevices) {
+            try {
+                def child = addChildButton()
+                child?.sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+            } catch (ex) {
+                log.error("Unable to create button device because the 'Child Button' DTH is not installed",ex)
             }
+        } else if (!state.createButtonEnabled && childDevices) {
+            removeChildButton(childDevices[0])
+        }
+    } else {
+        if (childDevices) {
+            removeChildButton(childDevices[0])
         }
     }
```

Comment:
Wouldn't it look better if You moved that part of the code to a separate function ?
That way You wouldn't have to call whole `initialize()` from `updated()`

Reply:
Oh, The function initialize() is a private function and only called by updated(). And the code from Line98~100 are moved before initialize() into initialize(). So the whole function initialize() should be called by updated().